### PR TITLE
Use COM smart pointers in WASAPI driver

### PR DESCRIPTION
**Pull request by lalitshankarch**
Use COM smart pointer (`ComPtr`) to simplify memory management in WASAPI driver.

This reduces chances of calling `Release()` when a COM object has not been allocated.

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks for the contribution!

Could you rebase your branch on latest upstream `master` to remove the merge commit? See https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase

**Comment by lalitshankarchowdhury** [comment]
> Thanks for the contribution!
> 
> Could you rebase your branch on latest upstream `master` to remove the merge commit? See https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase

Done.

File: drivers/wasapi/audio_driver_wasapi.cpp [hunk 1]
```diff
@@ -129,16 +123,11 @@ static bool default_input_device_changed = false;
 
 class CMMNotificationClient : public IMMNotificationClient {
 	LONG _cRef = 1;
-	IMMDeviceEnumerator *_pEnumerator = nullptr;
+	ComPtr<IMMDeviceEnumerator> _pEnumerator = nullptr;
```
[the remark this answers is not in the document]
Comment (lalitshankarchowdhury) [reply drivers/wasapi/audio_driver_wasapi.cpp]: Thanks for the heads up! Removed the dead code in my latest commit.

File: drivers/wasapi/audio_driver_wasapi.cpp [hunk 2]
```diff
@@ -537,10 +517,6 @@ Error AudioDriverWASAPI::audio_device_finish(AudioDeviceWASAPI *p_device) {
 		p_device->active.clear();
 	}
 
-	SAFE_RELEASE(p_device->audio_client)
-	SAFE_RELEASE(p_device->render_client)
-	SAFE_RELEASE(p_device->capture_client)
```
[the remark this answers is not in the document]
Comment (lalitshankarchowdhury) [reply drivers/wasapi/audio_driver_wasapi.cpp:542]: Thanks for the heads up! Fixed in my latest commit.

I added a `Reset()` method call for `audio_client`, `render_client`, and `capture_client` which will explicitly release the allocated resource.

And when destructor of `AudioDeviceWASAPI` is called, the reference count of these members will be set to 0 (if not already done, by calling `audio_device_finish`) therefore automatically releasing them.

**Review by bruvzg — APPROVED** [review]
Please squash your commits, see [pull request workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase).

**Comment by lalitshankarchowdhury** [comment]
@akien-mga @bruvzg All commits have been squashed and all checks have passed. Thank you for your time.

**Comment by lalitshankarchowdhury** [comment]
Fixed crash on Windows when `IAudioClient3` was being initialized.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**